File: 520. Detect Capital.py
#  https://leetcode.com/problems/detect-capital/  

class Solution:
    def detectCapitalUse(self, s: str) -> bool:
        if len(s) == 1:
            return True
        firstisuppper = False
        if s[0].isupper():
            firstisuppper = True
        if firstisuppper:
            
            if s[1].isupper():
                i = 2
                while i < len(s):
                    if s[i].islower():
                        return False
                    i += 1
                return True
            else:
                i = 2
                while i < len(s):
                    if s[i].isupper():
                        return False
                    i += 1
                return True
    
        else:
            i = 1
            while i < len(s):
                if s[i].isupper():
                    return False
                i += 1
            return True

File: 520. Detect Capital.cpp
/* https://leetcode.com/problems/detect-capital/  */

class Solution {
public:
    bool detectCapitalUse(string s) {
        if (s.size() == 1)
            return true;
        
        char first = s[0];
        bool ok = false;
        if (isupper(first))
            ok = true;
        
        if (ok)
        {
            if (isupper(s[1]))
            {
                int i = 2;
                while (i < s.size())
                {
                    if (islower(s[i]))
                        return false;
                    i++;
                }
                return true;
            }
            else
            {
                int i = 2;
                while (i < s.size())
                {
                    if (isupper(s[i]))
                        return false;
                    i++;
                }
                return true;
            }
        }
        else
        {
            int i = 1;
            while (i < s.size())
            {
                if (isupper(s[i]))
                    return false;
                i++;
            }
        }   return true;
    }
};
